databricks/notebooks/rapids_airline_hyperopt.ipynb

```diff
@@ -0,0 +1,419 @@
+    "\n",
+    "We'll use airlines data for this demo - the aim is to predict if an airline will be delayed or not using a target variable `ArrDelayBinary`. We'll use a Random Forest Classifier as the model for the learning task. \n",
+    "\n",
+    "We will compare the performance between the CPU version with scikit-learn to the GPU version with RAPIDS libraries."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "warnings.filterwarnings(\"ignore\", category=DeprecationWarning)\n",
+    "warnings.filterwarnings(\"ignore\", category=FutureWarning)\n",
+    "\n",
+    "import cudf\n",
+    "import cuml\n",
+    "\n",
+    "import mlflow\n",
+    "import hyperopt\n",
+    "\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
+    "import mlflow.sklearn\n",
+    "from mlflow.tracking.client import MlflowClient\n",
+    "from hyperopt import fmin, tpe, hp, Trials, STATUS_OK\n",
+    "\n",
+    "import cuml.ensemble\n",
+    "import cuml.metrics\n",
+    "import cuml.preprocessing.model_selection\n",
+    "\n",
+    "import sklearn.ensemble\n",
+    "import sklearn.metrics\n",
+    "import sklearn.model_selection"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preapring the data\n",
+    "\n",
+    "1. Download the data into your local system from [Airline 20M](https://rapids-csp.s3-us-west-2.amazonaws.com/data/airline_20000000.orc). This file is in ORC format, since Pandas can't easily read ORC files, we will convert the file to Parquet. \n",
+    "\n",
+    "2. In Databricks, upload the ORC file to your Databricks file system (DBFS). Read more about [DBFS CLI here](https://docs.databricks.com/data/databricks-file-system.html#databricks-cli). \n",
+    "\n",
+    "    From the local terminal use the `dbfs cp _local_path_/airline_20000000.orc dbfs:/_your_own_path/airline_20000000.orc` command to place the file in dbfs.\n",
+    "\n",
+    "3. Once it is in the DBFS, use the following code to convert it to Parquet."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Uncomment this line for generating the parquet file - run just once!\n",
+    "\n",
+    "# import cudf\n",
+    "# df = cudf.read_orc(\"/_your_own_path/airline_20000000.orc\")\n",
+    "# df.to_parquet(\"/_your_own_path/airline_20000000.parquet\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from contextlib import contextmanager\n",
+    "import time\n",
+    "\n",
+    "@contextmanager\n",
```

Comment:
Maybe document:
`# Small utility that times a block of code and prints how long it took to execute`